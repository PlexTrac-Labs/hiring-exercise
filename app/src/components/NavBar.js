import React from "react";
import { NavLink } from "react-router-dom";

function NavBar() {
  return (
    <div className="navigation">
      <nav className="navbar navbar-expand navbar-dark bg-dark">
        <div className="container">
          <NavLink className="navbar-brand" to="/">
            Hiring Exercise
          </NavLink>
          <div>
            <ul className="navbar-nav ml-auto">
              <li className="nav-item">
                <NavLink className="nav-link" to="/">
                  Log In
                  <span className="sr-only">(For now)</span>
                </NavLink>
              </li>
              <li className="nav-item">
                <NavLink className="nav-link" to="/profile">
                  Profile
                </NavLink>
              </li>
              <li className="nav-item">
                <NavLink className="nav-link" to="/userlist">
                  User List
                </NavLink>
              </li>
              <li className="nav-item">
                <NavLink className="nav-link" to="/resetpassword">
                  Reset Password
                </NavLink>
              </li>
              <li className="nav-item">
                <NavLink className="nav-link" to="/logout">
                  Log Out
                </NavLink>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  );
}

export default NavBar;
