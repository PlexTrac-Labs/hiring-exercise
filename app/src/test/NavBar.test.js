import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import NavBar from "../components/NavBar";

afterEach(cleanup);

//Mock the 3rd party react router dom
const mockNavLink = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  NavLink: () => mockNavLink
}));

it("renders correctly", async () => {
  const tree = renderer.create(<NavBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
