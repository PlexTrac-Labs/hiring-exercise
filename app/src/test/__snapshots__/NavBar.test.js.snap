// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="navigation"
>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <div
      className="container"
    >
      <div>
        <ul
          className="navbar-nav ml-auto"
        >
          <li
            className="nav-item"
          />
          <li
            className="nav-item"
          />
          <li
            className="nav-item"
          />
          <li
            className="nav-item"
          />
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
